command/import.go

```diff
@@ -178,6 +178,9 @@ func (c *ImportCommand) Run(args []string) int {
 		// We assume the same module as the resource address here, which
 		// may get resolved to an inherited provider when we construct the
 		// import graph inside ctx.Import, called below.
+		if rc != nil && rc.ProviderConfigRef != nil {
+			providerAddr = rc.ProviderConfigAddr().Absolute(addr.Module)
```

Comment:
In the `terraform import` docs for `-provider` it currently states the following:

> * `-provider=provider` - Specified provider to use for import. The value should be a provider alias in the form `TYPE.ALIAS`, such as "aws.eu". This defaults to the normal provider based on the prefix of the resource being imported. You usually don't need to specify this.

At the very least we should probably update the statement about the default to say that it consults the `provider` argument in the configuration.

However, I was also pondering whether we should reword this to be more definitive about the fact that this argument is now, in a sense, deprecated:

> `-provider=provider` - **Deprecated** Override the provider configuration to use when importing the object. By default, Terraform uses the provider specified in the configuration for the target resource, and that is the best behavior in most cases.

What do you think? :thinking:

Reply:
I like it! I've updated the cli usage and the website docs.